library/sn3218.py

```diff
@@ -17,6 +9,26 @@
 CMD_UPDATE = 0x16
 CMD_RESET = 0x17
 
+if sys.version_info < (3, ):
+    SMBUS = "python-smbus"
+else:
+    SMBUS = "python3-smbus"
+
+# Helper function to avoid exception chaining output in python3.
+# use exec to shield newer syntax from older python
+if sys.version_info < (3, 3):
+    def _raise_from_none(exc):
+        raise exc
+else:
+    exec("def _raise_from_none(exc): raise exc from None")
+
+try:
+    from smbus import SMBus
+except ImportError:
+    err_string = "This library requires %s\n" % SMBUS + \
```

Comment:
While `%` formatting is going to remain valid in Python for the foreseeable future, we much prefer the use of `.format()`. Even though it's slightly more verbose I think it's less baffling to beginners.

This could probably be:

```python
err_string = """This library requires {smbus}
Install with: sudo apt install {smbus}""".format(smbus=SMBUS)
```

Reply:
Changed to .format(). It looks weird with multi line out-dent, and slightly less weird as one long line (not the longest in the file), so I took the latter.

Reply:
Ping?